### ISSUE TITLE
filterPos incorrect when filterList contains object types

### ISSUE THREAD

**Issue by JeffJankowski** [gregnb/mui-datatables#1479]
## Expected Behavior
When `filterList` is populated with non-primitive object types, `customFilterListOptions.update: function(filterList, filterPos, index) => filterList` should provide a proper `filterPos` index to the custom update function.

My current use-case has a filter list with the following data shape for complex logic when fetching remote data:
```
interface Filter {
  label: string;
  value: string | number | boolean;
  key: string;
}

filterList: Filter[][]
```

I need `update` to return a correct `filterPos` to properly remove chips in a drill-down filtering scenario (eg. country -> state).

## Current Behavior
`filterPos` is always returning `-1`. I assume this is because of the naive equality check of reference types via `indexOf`:
https://github.com/gregnb/mui-datatables/blob/a382f3e8e850c7fcdd395f8594a5e0afefcaf905/src/MUIDataTable.js#L1372

I can't figure out a way around this issue without forking the repo (which I'd like to avoid). Possible solutions:
1. Add additional prop that allows me to set my own, custom comparator
1. Return filter `value` instead of array index (`filterPos`) to do my own comparison logic in the `update` function
1. Deep-object equality check instead of naive reference equality (probably not ideal)

## Steps to Reproduce (for bugs)
I can provide a codesandox environment if requested.

## Your Environment

| Tech         | Version |
|--------------|---------|
| Material-UI  |    4.10.2     |
| MUI-datatables  |    3.3.1     |
| React        |    16.3.1     |
| browser      |         |
| etc          |         |

**Comment by patorjk** [comment]
Can you provide a codesandbox? 

I've been terribly busy recently, I would accept a PR to fix this.

**Comment by JeffJankowski** [comment]
@patorjk PR created: #1481

**Comment by JeffJankowski** [comment]
Merged into master in v3.4.1 #1487
